### DISCUSSION TITLE
pip-compile: Suddenly enforcing .txt in fileMatch regex

### DISCUSSION THREAD

**Discussion by peschmae** [renovatebot/renovate#27559] [Request Help] [answered]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us what version of Renovate you run.

37.214

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Describe the problem

I've recently upgraded renovate in our setup from 37.73 to 37.214 and then it opened a pull request to migrate the renovate.json.

We are using `*.in` files for our python requirements, that should be recompiled with pip-compile, and that work like a charm, but renovate is now adamant about replacing the `in` with `txt` in the configuration, and is also substituting it during the run.

I saw that there was a change for pip-compile requirements in subdirectories that was merged in 37.203, so I reverted to 37.202 but this is already happening there.

The [docs](https://docs.renovatebot.com/modules/manager/pip-compile/#non-configured-filematch) mention that the pip-compile manager has no fileMatch configured, so I don't think this should be enforced.

Reproduction repo: https://github.com/peschmae/renovate-reprod-pip-compile-filematch

### Relevant debug logs

<details><summary>Logs</summary>

```
DEBUG: Config migration necessary
{
  "oldConfig": {
    "$schema": "https://docs.renovatebot.com/renovate-schema.json",
    "extends": [
      "mergeConfidence:all-badges",
      "config:recommended"
    ],
    "pip-compile": {
      "fileMatch": [
        "(^|/)requirements\\/.*\\.in$"
      ]
    }
  }
  "newConfig": {
    "$schema": "https://docs.renovatebot.com/renovate-schema.json",
    "extends": [
      "mergeConfidence:all-badges",
      "config:recommended"
    ],
    "pip-compile": {
      "fileMatch": [
        "(^|/)requirements\\/.*\\.txt$"
      ]
    }
  }
}
```
Log output from renovate app, job # bed2c6b3-aba8-4198-a6b0-cf2a766e9ff6
</details>

### Have you created a minimal reproduction repository?

I have linked to a minimal reproduction in the description above

**Comment by viceice** [comment]
intentional behavior. you now need to match the output files instead of the input.

read the release notes from renovate.

**Comment by peschmae** [comment, marked as answer]
Introduced in https://github.com/renovatebot/renovate/commit/2df7ef51aa898ec3a13d8e1ef87c6b2495e4c358

Leaves a documentation issue, since the documentation references `*.in` files, and the fileMatching section doesn't mention that output files should be matched

**Reply by viceice** [reply]
Where did you read this? the docs are already updated: https://docs.renovatebot.com/modules/manager/pip-compile/#non-configured-filematch

**Reply by peschmae** [reply]
Yes, I fixed them: https://github.com/renovatebot/renovate/releases/tag/37.219.1
